### PR TITLE
Rek eclkc cloudgov

### PR DESCRIPTION
**Pull request by kilbergr**
# [ECLKC Cloudgov attempt](https://ocio-jira.acf.hhs.gov/browse/OHSH-315)

Changes proposed in this pull request:

- Add a research doc for eclkc cloud.gov attempt.

### CODE REVIEW

File: docs/research/applications/eclkc-cloudgov.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
+# ECLKC Cloud.Gov Feasibility
+
+Rebecca Kilberg
+11/24/2021
+
+## Context
+
+We wanted to determine how difficult it would be to get various applications on cloud.gov due to [the benefits it offers around compliance, security, and best practices](../design-docs/cloud-dot-gov-feasibility). We selected the ECLKC app because we believed it would be the most difficult to migrate to cloud.gov since it uses the greatest number of services and is a Drupal app, which we were aware had some historical challenges getting onto cloud.gov. These challenges include the way Drupal depends on the local filesystem (incompatible with cloud.gov) for configuration and that the cloud.gov provided alternative was to use S3, which previous dev teams had struggled to use with Drupal.
```
Comment (reid-truss) [inline docs/research/applications/eclkc-cloudgov.md:8]: do we have more context on "which previous dev teams had struggled to use with Drupal" re: s3? if I was reading this without additional context, I'd definitely wonder why they struggled with it, so any little bit of info we can drop there would be helpful

File: docs/research/applications/eclkc-cloudgov.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+* Create a production RDS DB instance and pass the settings in `web/sites/defaults/settings.php`. We could use the following command to create a db with the fields of interest set:
+
+    ```[bash]
+    CREATE DATABASE db CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
+
+    GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, DROP, INDEX, ALTER, CREATE TEMPORARY TABLES ON db.* TO 'db'@'%' IDENTIFIED BY 'db';
+    ```
+
+* Take the data (`production_full.sql`) from SFTP server that is synced every night.
+* Specify path of static files in the `web/sites/defaults/ettings.php` in the public path.
```
Comment (reid-truss) [inline docs/research/applications/eclkc-cloudgov.md:38]: is this meant to be `web/sites/defaults/settings.php`?

File: docs/research/applications/eclkc-cloudgov.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+```[bash]
+  ERROR 1053 (08S01) at line 27617: Server shutdown in progress
+```
+
+which I dealt with simply by rerunning the command.
+
+## Death Knell
+
+The issue we're currently facing is that there is something off with the routing. We've made it to the point of encountering the following error when visiting our cloud.gov site (hsicc.app.cloud.gov):
```
Comment (reid-truss) [inline docs/research/applications/eclkc-cloudgov.md:170]: might be good to specify routing within drupal/php/the app (routing in my brain immediately jumps to network routing, so depending on audience, they may assume something other than what you intended)

**Review by reid-truss — COMMENTED** [review]
just a small number of clarifications/questions and maybe one typo - otherwise LGTM, thanks for writing this all up.

File: docs/research/applications/eclkc-cloudgov.md (hunk 1 above)
Comment (kilbergr) [reply docs/research/applications/eclkc-cloudgov.md:8]: I've added the info I have but don't have all that much more unfortunately.

**Review by reid-truss — APPROVED** [review]

**Review by ssatagopan — COMMENTED** [review]
Any lessons learned that could be applicable to other apps as well?

File: docs/research/applications/eclkc-cloudgov.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+## Accessing cloud.gov
+
+To begin, we had to get access to cloud.gov. We relied on our partner at 18F, Nate Price, to get us access. As someone with a .gov email address, he was able to sign up for a [free sandbox space](https://cloud.gov/docs/pricing/free-limited-sandbox/). He then granted the rest of us access. At this point, we realized no one had permission to create an organization for us to use, and that the free sandbox space was too small for us to set up anything meaningful.
+
+18F partner Ryan Ahearn negotiated with TTA SmartHub, who already had a cloud.gov account, to allow us to use their organization to set up our own space. We ended up being given access to the SmartHub organization (`hhs-acf-ohs-tta`) with a 2GB memory allocation for our exploration. The space we were given is called `eclkc-investigation`.
+
+Once we had permission to access cloud.gov, I installed [cloudfoundry CLI (7)](https://docs.cloudfoundry.org/cf-cli/install-go-cli.html#pkg-mac) and was able to [access the web GUI and use cf on the command line](https://cloud.gov/docs/getting-started/setup/).
```
Comment (eeeady) [inline docs/research/applications/eclkc-cloudgov.md:16]: for clarity, did you install this locally or on an operations machine in our AWS envrionment?

File: docs/research/applications/eclkc-cloudgov.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+## Standing up the server
+
+You can use cloud.gov in one of two ways: containerize your app outside of cloud.gov and run it inside cloud.gov or push your app code in cloud.gov and have the container built for you. Either way, the apps well-suited to cloud.gov can run in stateless container.
```
Comment (eeeady) [inline docs/research/applications/eclkc-cloudgov.md:20]:
```suggestion
You can use cloud.gov in one of two ways: containerize your app outside of cloud.gov and run it inside cloud.gov or push your app code in cloud.gov and have the container built for you. Either way, the apps well-suited to cloud.gov must run in stateless container.
```

**Review by eeeady — COMMENTED** [review]
Generally a good and thorough summary of the investigation.

I had a couple of little things in line and I also think it would be good to have a short summary of our conclusions at this time at the top.

File: docs/research/applications/eclkc-cloudgov.md (hunk 4 above)
Comment (kilbergr) [reply docs/research/applications/eclkc-cloudgov.md:16]: locally. I'll add that and that I installed in Ansible/Nagios later.

**Review by ssatagopan — APPROVED** [review]
LGTM. On to the next app!

**Pull request merged**